pkgs/development/python-modules/pyqtwebengine/default.nix

```diff
@@ -29,12 +30,18 @@ in buildPythonPackage rec {
   nativeBuildInputs = [
     pkg-config
     qmake
+  ] ++ lib.optionals (stdenv.buildPlatform == stdenv.hostPlatform) [
     sip
+  ] ++ lib.optionals (stdenv.buildPlatform != stdenv.hostPlatform) [
+    buildPackages.python3Packages.sip
```

Comment:
this will not work when cross-compiling a non-default python version

needs to use `python.pythonForBuild.pkgs.sip`

Reply:
> needs to use `python.pythonForBuild.pkgs.sip`

`python.pythonForBuild` is `pkgsBuildBuild.python`, but we want `pkgsBuildHost.python`.  I will try to find a way to get that via `python.*`.

Reply:
> > needs to use `python.pythonForBuild.pkgs.sip`
> 
> `python.pythonForBuild` is `pkgsBuildBuild.python`, but we want `pkgsBuildHost.python`. I will try to find a way to get that via `python.*`.

Sip does c stuff so I guess target matters for it?

Reply:
https://github.com/NixOS/nixpkgs/pull/265570

Reply:
> Sip does c stuff so I guess target matters for it?

Quite likely.  Or even using `pkg-config` would make it target-sensitive.